src/devices_models/devices/common/cost_functions.jl

```diff
@@ -49,6 +49,17 @@ function AddCostSpec(
     error("AddCostSpec is not implemented for $T / $U")
 end
 
+function add_service_variables!(
+    spec::AddCostSpec,
+    services::Vector{T},
+) where {T <: PSY.Service}
+    for service in services
+        name = PSY.get_name(service)
+        spec.addtional_linear_terms[name] = make_variable_name(name, typeof(service))
```

Comment:
It looks like this uses the fallback `make_variable_name` method where no VariableType is specified.  Should there be a new method, perhaps using ActiveServiceVariable?

Reply:
For this variable there isn't a pre-defined name (like the const names) it used the name of the service itself to build for the variable name in reserves.jl, so the new method would need use to pass the service along  with the variable struct, not sure if that's the best way to do this ?
https://github.com/NREL-SIIP/PowerSimulations.jl/blob/37e14409d543ba57e80632b53a1e6f2b8d8397b9/src/services_models/reserves.jl#L8-L14